crates/matrix-sdk/src/widget/capabilities.rs

```diff
@@ -50,6 +53,17 @@ pub struct Capabilities {
     pub requires_client: bool,
 }
 
+impl Capabilities {
+    /// Tells if a given raw event matches the read filter.
+    pub fn raw_event_matches_read_filter(&self, raw: &Raw<AnyTimelineEvent>) -> bool {
+        let Ok(filter_in) = raw.deserialize_as::<MatrixEventFilterInput>() else {
+            return false;
+        };
```

Comment:
We should log the deserialization error in case `deserialize_as` returns one, as before in `machine/mod.rs`.

Reply:
Done.